src/main/java/org/reaktivity/nukleus/http/internal/routable/stream/TargetInputEstablishedStreamFactory.java

```diff
@@ -417,7 +454,13 @@ private void decodeBufferedData()
                 streamState = this::handleStreamWhenNotBuffering;
                 if (endDeferred)
                 {
-                    doCleanup();
+                    connection.persistent = false;
+                    if (contentRemaining > 0)
+                    {
+                        // TODO: write an abort frame rather than end
+                        target.doEnd(targetId);
```

Comment:
Suggest adding `target.doAbort(...)` but implementing it to send `END` for now.
This will help you to identify the semantics correctly in the code where `ABORT` will be more appropriate.